### PR TITLE
MAINT: Remove docstring'd-out traceback for code that no longer raises

### PR DESCRIPTION
**Pull request by jbrockmendel**
with test.

Separate the formerly-raising code out from `__main__` block into `example_fdr_bonferroni`, implement a smoketest to check it runs without raising.

The new function example_fdr_bonferroni is cleaned up, the rest left as is.

### CODE REVIEW

File: statsmodels/sandbox/stats/multicomp.py [hunk 1]
```diff
@@ -1802,49 +1836,10 @@ def set_remove_subs(ssli):
         x = np.array([[0,0,1]]).T + np.random.randn(3, 20)
```
Comment (bashtage) [inline statsmodels/sandbox/stats/multicomp.py]: @pytest.mark.smoke
test_tukey_threegene()
...


and then remove main?

**Comment by josef-pkt** [comment]
don't merge this !

There should be no example functions and unit tests in modules (outside of examples and tests)
I don't think we want pytest outside of test modules, (pytest should be optional outside of running unit tests, isn't it?)
The entire main should be moved to an example file or even better to a notebook.

**Comment by bashtage** [comment]
I hadn't noticed this wasn't a test file. Yes pytest only in tests, and tests only in test.

**Comment by jbrockmendel** [comment]
> The entire main should be moved to an example file or even better to a notebook.

Would this actually be allowed?  Historically the suggestion of moving code between files has led to torches and pitchforks.

If so, then I'll _gladly_ make the suggestion change.

**Comment by josef-pkt** [comment]
We/I moved most of the main blocks in modules outside of the sandbox into example files, or deleted it if there was already an equivalent example.

In the sandbox it's different, main blocks can stay if there is still work to be done directly for the functions in the module. 
e.g. many sandbox.distribution modules have main blocks with the relevant examples and it's not helpful to move them to example files. They should be moved as part of substantive work and not as part of cleanup.

**Comment by jbrockmendel** [comment]
OK.  Can you please re-open and I'll try this again?

**Comment by jbrockmendel** [comment]
Updated as discussed, also moved `test_tukey_pvalues` to the new test_multicomp file.

**Comment by bashtage** [comment]
Looks like it takes @josef-pkt comments on board.

**Comment by jbrockmendel** [comment]
Comments have been addressed

**Comment by jbrockmendel** [comment]
Comments have been addressed

**Comment by bashtage** [comment]
Thanks.

**Pull request merged**